### DISCUSSION TITLE
Is there a way to decompile scripts with serializefields but without compiler errors

### DISCUSSION THREAD

**Discussion by lastone-hikiray** [AssetRipper/AssetRipper#858] [Q&A]
In most cases when I rip the game, I just want to go in the scenes/prefabs and watch scripts names and serializefields, but with compiler errors they are not loaded.

It will be great if we have option in scripts export format : "without compiler errors", which would remove method bodies and constructors from scripts after ILSpy decompilation.

**Comment by ds5678** [comment]
That's not going to happen. I considered this a long time ago, but the issues remain the same. Interfaces and abstract base methods would cause compiler errors, which defeats the purpose of doing this. As such, I opted for method stubbing instead of method stripping.

**Comment by dukemagus** [comment]
It's still possible to retrieve part of the metadata of the serializefield
and put it in a stub text file or in a comment at the end of the the
script, right?



Em sex., 16 de jun. de 2023 08:59, ds5678 ***@***.***>
escreveu:

**Reply by ds5678** [reply]
No, ILSpy controls the entire file text.
